CONTRIBUTING.md

```diff
@@ -435,68 +193,460 @@ The top-level repository directories are:
+
+The source format for all of these tools is plain text.
+This allows managing the documentation source files with git.
+
+## Writing Documentation
+
+There are several practices that we attempt to follow with
+documentation sources.
+These are intended to help restrict diffs to the text actually
+changed, which make diffs smaller and easier to read, which in turn
+makes changes easier to review (and easier to merge) and considerably
+reduces development friction.
+
+1. Don't generate long lines.
+   Have your text editor word-wrap at somewhere between 60 and 70 columns.
+
+2. New sentence, new line.
+   When starting a new sentence, begin it on a new line.
+   Don't reflow whole paragraphs; reflow only single sentences.
+
+3. Avoid making changes to lines you aren't actually making substantive
+   edits to.
+   Sometimes this means introducing an extra line break or two where
+   you otherwise might reflow.
+
+4. Commit whitespace and formatting (source formatting/layout) changes
+   separately from content changes.
+   (Thus, if you end up with too many extra line breaks or the like,
+   such that things get ugly or unreadable, fix it up as a separate
+   commit.)
+   Document such changes in their commit messages to help reviewers
+   skip over them.
 
-## Using Mergify
+5. Embedded links can be long and not play nicely with word-wrap; if
+   one gets ugly, put it on its own line.
+
+If you find that your editor is violating these principles for you
+(e.g. refusing to let you have reasonable-length lines, or reflowing
+text without being asked) and you can't figure out how to stop it, ask
+for help.
+
+A couple tips for writing effective Markdown:
+
+- You need a blank line before the first entry of a list, and between
+  entries.
+  If you leave the initial blank line off, strange things happen.
+
+- The whole of each list item should be indented past the bullet point.
+  This normally doesn't matter if you have a single paragraph, but if
+  you want multiple paragraphs or have other markup it can be important.
+
+## Updating Prebuilt Copies
+
+For some of the large documents (currently without much intentionality
+or pattern) a prebuilt PDF file is included in the repository.
+This is for the use of people browsing in their local checked-out
+clones of the tree, or on GitHub, who may not want to (or be able to,
+respectively) build the documentation themselves.
+These copies should be updated whenever any of their sources change,
+using the following procedure:
+
+1. Commit all the sources.
+
+2. When necessary (for changes that affect text that propagates via
+the SAW executable, such as the doc strings for the SAW intrinsics),
+rebuild SAW itself.
+
+3. Rebuild the PDF.
+
+4. Commit the updated PDF with a commit message along the lines of
+   "Regenerate the prebuilt manual".
+
+This makes sure that all the ingredients are fully up to date and that
+any and all metadata built into the PDF reflects the commit that
+changed the text.
```

Comment:
Thanks very much for writing up this detailed section about documentation! I will be sure this gets appropriately updated (like earlier comments, likely as part of / after #2202).